tests/run/generic/Enum.scala

```diff
@@ -0,0 +1,18 @@
+package generic
+
+import Shapes.Singleton
+
+trait Enum {
+  def enumTag: Int
```

Comment:
What use cases do you envision for `enumTag`? Back then, @densh and I planned to use tags to speed up pattern matching. Do you think that'd make sense?